kuksa-client/kuksa_client/grpc/__init__.py

```diff
@@ -562,7 +606,10 @@ def __enter__(self):
             channel = grpc.insecure_channel(self.target_host)
         self.channel = self.exit_stack.enter_context(channel)
         self.client_stub = val_pb2_grpc.VALStub(self.channel)
-        if self.ensure_startup_connection:
+        if self.authorization_header is None:
+            logger.debug(
+                "Can not ensure startup connection without token to authorize")
+        elif self.ensure_startup_connection:
             logger.debug("Connected to server: %s", self.get_server_info())
```

Comment:
If you end up here you will still get a `'No auth token provided'` error as `get_server_info()` does not care if `self.authorization_header` is defined. So if you request a client like this:

```
self._grpc_client = self._exit_stack.enter_context(kuksa_client.grpc.VSSClient(
                 host=self._ip,
                 port=self._port,
                 token = token_content,
                 #ensure_startup_connection = False
            ))
```

I.e. with token defined and `self.ensure_startup_connection` set.

If you add:

```
rpc_kwargs["metadata"] = self.generate_metadata_header(
            rpc_kwargs.get("metadata"))
```

... to  `get_server_info()` that use-case seems to work as well.

Then I can sort of agree - does  `get_server_info()` really need to request a token file. Should that not be possible to run without? But as long as it request it we need to provide it as I see it.